test/test_shadowState.py

```diff
@@ -0,0 +1,39 @@
+from unittest import TestCase
+
+from awsiot.iotshadow import ShadowState
+
+
+class ShadowTest(TestCase):
+
+    def test_shadow_state_payload_filled(self):
+        testState = ShadowState.from_payload({
+            "reported": {"Color": "Red"},
+            "desired": {"Color": "Blue"}
+        })
+        compareState = ShadowState(
+            reported={"Color": "Red"},
+            desired={"Color": "Blue"}
+        )
+        self.assertTrue(testState.to_payload() == compareState.to_payload())
+
+    def test_shadow_state_payload_partial_filled(self):
+        testState = ShadowState.from_payload({
+            "reported": {"Color": "Red"}
+        })
+        compareState = ShadowState(
+            reported={"Color": "Red"}
+        )
+        self.assertEqual(testState.to_payload(), compareState.to_payload())
```

Comment:
trivial: if there was a bug in to_payload(), this test would still pass, because we're just testing that both classes have the same output from their to_payload() function.

better to assert the to_payload() output against raw data

I mean, it's fine to also have an assert like this that that their output is the same, but what we really care about is that it is the data we expect